src/forms/subscriptions.ts

```diff
@@ -0,0 +1,401 @@
+        if (!SubscriptionFields.ConditionsCheckBoxFields.includes(condition.field)) {
+            this.unsupportedFields.push(condition.field);
+        }
+    }
+
+    // validateFieldOperator validates the trigger operator is supported by the app
+    validateFieldOperator(condition: any): void {
+        if (condition.operator !== 'changed') {
+            this.unsupportedOperators.push(`${condition.field} - ${condition.operator}`);
+        }
+    }
+
+    // getAnyConditions returns an array of Zendesk ANY trigger conditions
+    getAnyConditions(): any[] {
+        if (this.getSelectedSubTrigger() && this.getSelectedSubTrigger().conditions) {
+            return this.getSelectedSubTrigger().conditions.any;
+        }
+        return [];
+    }
+
+    // getAnyConditions returns an array of Zendesk ALL trigger conditions
+    getAllConditions(): any[] {
+        if (this.getSelectedSubTrigger() && this.getSelectedSubTrigger().conditions) {
+            return this.getSelectedSubTrigger().conditions.all;
+        }
+        return [];
+    }
+
+    // addErrorMessageField adds a text field with a message when a trigger has
+    // conditions not supported by the app
+    addErrorMessageField(link: string): void {
+        let text = 'The following condition fields are not currently supported by the app. Please visit the trigger link to modify the conditions for this subscription';
+        text += '\n\n';
+        text += this.getBulletedList('Unsupported Fields', this.unsupportedFields);
+        text += '\n\n';
+        text += this.getBulletedList('Unsupported Field Operators', this.unsupportedOperators);
+        text += '\n\n' + link;
+
+        // TODO This message is better suited as an error message next to buttons
+        // Need to add clickable link to trigger.  Always show the link (next to Subscription Name)
+        // disble the submit button';
+        const f: AppField = {
+            name: SubscriptionFields.UnsupportedFieldsText_Name,
+            type: AppFieldTypes.TEXT,
+            subtype: 'textarea',
+            label: 'Optional message',
+            value: text,
+            readonly: true,
+        };
+
+        this.builder.addField(f);
+    }
+
+    // getBulletedList returns a bulleted list of items with options header
+    // pretext
+    getBulletedList(pretext: string, items: string[]): string {
+        let text = '* ' + items.join('\n* ');
+        if (pretext) {
+            text = `###  ${pretext}\n` + text;
+        }
+        return text;
+    }
+
+    // isZdFieldChecked returns a boolean representing if a value in the saved
+    // Zendesk trigger is true or false
+    isZdFieldChecked(conditions: any, name: string): boolean {
+        const condition = conditions.filter(this.byName(name));
+        return condition.length === 1;
+    }
+
+    // byName is a map filter function to retrieve a given fieldName from an
+    // array of conditions
+    byName(name: string): boolean {
```

Comment:
It is a zendesk field.  The problem is that zendesk types aren't defined for many objects.  I added one for this case though because it's small.  also used locally, so I added it to this file.